cterasdk/core/files/collaboration.py

```diff
@@ -0,0 +1,156 @@
+import logging
+from . import common
+from ..enum import ProtectionLevel, CollaboratorType, SearchType, PortalAccountType, FileAccessMode
+from ..types import UserAccount, GroupAccount
+from ...common import Object
+from ...exception import CTERAException
+
+
+def _search_collaboration_member(ctera_host, account, cloud_folder_uid):
+    search_param = Object()
+
+    if account.account_type == PortalAccountType.User:
+        search_param.searchType = SearchType.Users
+    elif account.account_type == PortalAccountType.Group:
+        search_param.searchType = SearchType.Groups
+    else:
+        raise CTERAException("Invalid account type", None, account_type=account.account_type)
+
+    search_param.searchTerm = account.name
+    search_param.resourceUid = cloud_folder_uid
+    search_param.countLimit = 1
+    response = ctera_host.execute('', 'searchCollaborationMembers', search_param)
+    if not response.objects:
+        logging.getLogger().warning('Could not find results that match your search criteria. %s', {'name': account.name})
+    elif len(response.objects) > 1:
+        logging.getLogger().warning('Search returned multiple results. Skipping...')
+    else:
+        collaborator = response.objects.pop()
```

Comment:
Could it happen that you got here and both "if"s return false? If yes, you should add a warning log similar to the previous cases. If not, then the if should only affect the print and you can return once outside of it

Reply:
Fixed to address all possible scenarios.